tests/suites/test_suite_ctr_drbg.function

```diff
@@ -126,58 +75,111 @@ void ctr_drbg_validate( int reseed_mode, char *nonce_string,
          *                 reseed[:reseed_len]) */
         TEST_ASSERT( mbedtls_ctr_drbg_reseed(
                          &ctx,
-                         reseed, reseed_len ) == 0 );
+                         reseed->x, reseed->len ) == 0 );
     }
 
-    /* CTR_DRBG_Generate(result_len * 8 bits, add2[:add2_len]) -> buf */
+    /* CTR_DRBG_Generate(result->len * 8 bits, add2->x[:add2->len]) -> buf */
     /* Then reseed if prediction resistance is enabled. */
     TEST_ASSERT( mbedtls_ctr_drbg_random_with_add(
                      &ctx,
-                     buf, result_len,
-                     add2, add2_len ) == 0 );
-    TEST_ASSERT( memcmp( buf, result, result_len ) == 0 );
+                     buf, result->len,
+                     add2->x, add2->len ) == 0 );
+    TEST_ASSERT( memcmp( buf, result->x, result->len ) == 0 );
 
 exit:
     mbedtls_ctr_drbg_free( &ctx );
+    return 0;
+}
+
+/* END_HEADER */
+
+/* BEGIN_DEPENDENCIES
+ * depends_on:MBEDTLS_CTR_DRBG_C
+ * END_DEPENDENCIES
+ */
+
+/* BEGIN_CASE */
+void ctr_drbg_special_behaviours( )
+{
+    mbedtls_ctr_drbg_context ctx;
+    unsigned char output[512];
+    unsigned char additional[512];
+
+    mbedtls_ctr_drbg_init( &ctx );
+    memset( output, 0, sizeof( output ) );
+    memset( additional, 0, sizeof( additional ) );
+
+    TEST_ASSERT( mbedtls_ctr_drbg_random_with_add( &ctx,
+                        output, MBEDTLS_CTR_DRBG_MAX_REQUEST + 1,
+                        additional, 16 ) ==
+                        MBEDTLS_ERR_CTR_DRBG_REQUEST_TOO_BIG );
+    TEST_ASSERT( mbedtls_ctr_drbg_random_with_add( &ctx,
+                        output, 16,
+                        additional, MBEDTLS_CTR_DRBG_MAX_INPUT + 1 ) ==
+                        MBEDTLS_ERR_CTR_DRBG_INPUT_TOO_BIG );
+
+    TEST_ASSERT( mbedtls_ctr_drbg_reseed( &ctx, additional,
+                        MBEDTLS_CTR_DRBG_MAX_SEED_INPUT + 1 ) ==
+                        MBEDTLS_ERR_CTR_DRBG_INPUT_TOO_BIG );
+
+    mbedtls_ctr_drbg_set_entropy_len( &ctx, ~0 );
+    TEST_ASSERT( mbedtls_ctr_drbg_reseed( &ctx, additional,
+                        MBEDTLS_CTR_DRBG_MAX_SEED_INPUT ) ==
+                        MBEDTLS_ERR_CTR_DRBG_INPUT_TOO_BIG );
+exit:
+    mbedtls_ctr_drbg_free( &ctx );
+}
+/* END_CASE */
+
+/* BEGIN_CASE */
+void ctr_drbg_validate( int reseed_mode, data_t * nonce,
```

Comment:
Please remove this function. In the test data, replace `ctr_drbg_validate:RESEED_NEVER` by `ctr_drbg_validate_no_reseed`, `ctr_drbg_validate:RESEED_ALWAYS` by `ctr_drbg_validate_pr`, and `ctr_drbg_validate:RESEED_FIRST` by `ctr_drbg_validate_reseed_first` (some subsequent arguments may need to be adjusted). Write `ctr_drbg_validate_reseed_first` on the same model as the other three. Rename `ctr_drbg_validate_nopr` to `ctr_drbg_validate_reseed_between` for consistency.

Reply:
I've made the changes requested (thanks for the detailed explanation)